From 637f5afe18896b99cccdd00e8972941548748e79 Mon Sep 17 00:00:00 2001
From: Meriipu <Meriipu@users.noreply.github.com>
Date: Mon, 12 Oct 2020 02:19:24 +0200
Subject: [PATCH 5/9] skip the topsites provider due to horrible
 priority-stealing behaviour

even though history was getting sent to the query this provider seems
to somehow have dropped it and returned NOTHING because topsites are
disabled in my setup, leading to the dropmarker doing absolutely
nothing.

With the topsites provider removed, firefox is forced to use the
history one or whatever it is called.
---
 browser/components/urlbar/UrlbarProvidersManager.jsm | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/browser/components/urlbar/UrlbarProvidersManager.jsm b/browser/components/urlbar/UrlbarProvidersManager.jsm
index 0b3d015113b8..a286e72f8fc0 100644
--- a/browser/components/urlbar/UrlbarProvidersManager.jsm
+++ b/browser/components/urlbar/UrlbarProvidersManager.jsm
@@ -353,6 +353,13 @@ class Query {
       //     return;
       //   }
       provider.queryInstance = this;
+
+      // Christ almighty
+      if (provider.name == "UrlbarProviderTopSites")
+      {
+        console.log("[UrlbarProvidersManager] Skipping provider: " + provider.name + " " + provider.type);
+        continue;
+      }
       activePromises.push(
         // Not all isActive implementations are async, so wrap the call in a
         // promise so we can be sure we can call `then` on it.  Note that
-- 
2.28.0

